src/client-side-metrics/operation-metrics-collector.ts

```diff
@@ -0,0 +1,337 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as fs from 'fs';
+import {IMetricsHandler} from './metrics-handler';
+import {MethodName, StreamingState} from './client-side-metrics-attributes';
+import {grpc} from 'google-gax';
+
+/**
+ * An interface representing a Date-like object.  Provides a `getTime` method
+ * for retrieving the time value in milliseconds.  Used for abstracting time
+ * in tests.
+ */
+interface DateLike {
+  /**
+   * Returns the time value in milliseconds.
+   * @returns The time value in milliseconds.
+   */
+  getTime(): number;
+}
+
+/**
+ * An interface representing a tabular API surface, such as a Bigtable table.
```

Comment:
why does this exist here? Isn't there [already a class for the TabularAPISurface](https://github.com/googleapis/nodejs-bigtable/blob/5da73e1eec10a4729d158ebcc181df88497ea5c2/src/tabular-api-surface.ts#L155)?

Reply:
An `ITabularApiSurface` parameter is provided when the metrics collector is created. We don't want to require this parameter to be a whole `TabularApiSurface` object because only a small subset of the properties are actually used and we don't want to restrict the `OperationMetricsCollector` further than we have to. In particular, this simplifies the test for the metrics collector where we create a simple `FakeTable` object which would otherwise have to be a whole `TabularApiSurface` object.